feat: write user task migration guide

christian-konrad opened the pull request:
## Description

<!-- This helps the reviewers by providing an overview of what to expect in the PR. Linking to an issue is preferred but not required. -->
<!-- Add an assignee and use component: labels for good hygiene! -->

Implements #3434

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->
<!-- Help the DevEx team prioritize our work (reviews, merges, etc.) by opening PRs sooner. -->

- [x] This change is not yet live and should not be merged until March 12th.
- [x] This change or page is part of a marketing blog, conference talk, or something else on a schedule.

## PR Checklist

<!-- Keep in mind, Camunda maintains 18 months of versions. Backporting your change or including it in multiple versions is common. -->

- [x] I have added changes to the main `/docs` directory (aka `/next/`).
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer.

christian-konrad commented:
@pepopowitz @christinaausley @akeller I didn't know who's best to invite to a review among you, so please one of you pick it as it fits best :)

christian-konrad commented:
Can someone help me run prettier? For some reason, it's broken on my machine (and helping me fix it would be great as well :) )

akeller commented:
> Can someone help me run prettier? For some reason, it's broken on my machine (and helping me fix it would be great as well :) )

@christian-konrad I got you 😄 

Can you run `npm run format`? If you get an error, connect with us on #ask-c8-documentation. And we'll debug further.

I did notice that this file wasn't being picked up by `npm run format` so in my IDE I had to format this file specifically. Is that what you mean by prettier not working for you?

akeller commented:
Sorry for the thrash on this PR - https://camunda.slack.com/archives/C026U8GBNSW/p1709831600253279

We'll get a PR merged and I'll clean this up, formatting and all.

akeller commented:
@pepopowitz please review this PR - it has js, jsx, and css changes that I'd like you to be aware of as our DRI of docs infra.

pepopowitz commented on docs/guides/migrate-to-zeebe-user-tasks.md:
```diff
@@ -0,0 +1,376 @@
+
+Use the following decision helper questionaire to figure out what's right for you.
+
+<FormViewer schema={ userTaskMigrationDecisionHelperForm } customStyles={{
+    border: "none",
+    padding: "0px",
+    maxWidth: "unset",
+    background: "transparent",
+    width: "calc(100% + 42px)",
+    marginLeft: "-28px"
+}} />
```
Blocking: 

Maybe it's still in progress, but I don't see this component rendering when I run locally. I think there might be an issue with the configuration for it -- in its current state, I see nothing the whole time; when I edit the related userTaskMigrationDecisionHelperForm.js to have an empty `components: []` array, I at least see the "Loading..." placeholder flash on load. 

This is what it looks like right now when I run locally: 

<img width="760" alt="image" src="https://github.com/camunda/camunda-docs/assets/1627089/74615463-ed1b-43bb-863e-a13cb0c08800">

christian-konrad replied:
Have you run npm install before? It requires an update of form-js

pepopowitz commented on docs/guides/migrate-to-zeebe-user-tasks.md:
```diff
@@ -0,0 +1,376 @@
+
+You can also operate both task types at the same time in the same application by utilizing both APIs. We recommend this for a smooth migration, but you should eventually update all processes to use the new task type to use all benefits. The following image illustrates how to route API calls to the respective APIs:
+
+<img src={APIArchitectureImg} className={styles.noShadow} style={{width: 800}} alt="Task API Architecture" />
```
Blocking: 

The images on this page have the shadow scrubbed, like the image below. [We previously found that images without any border or shadow were confusing](https://github.com/camunda/developer-experience/issues/24), so we added a shadow. @christian-konrad, you opened #2059 to remove the _shadow_ (but not the border); we have not moved on that issue, because the border vs outline debate seems to me to be subjective opinion. 

I would be fine with these images removing the _shadow_ without also removing the outline, but removing both is a regression in terms of readability, supported by the feedback we originally received for images with no outline. 

cc @christinaausley and @akeller because I think they probably also have opinions about the presence of borders on images. 

<img width="865" alt="image" src="https://github.com/camunda/camunda-docs/assets/1627089/c278db46-9689-457d-9365-8b22c0cc0786">

christian-konrad replied:
I can readd the border; the drop-shadow just decreases readability in my opinion and elevates the images were it's not needed, but I agree to the border.

pepopowitz commented on docs/guides/assets/css/cleanTable.module.css:
non-blocking: 

Is this styling that we want to apply to all tables? If it is, then we should apply it globally instead of only on this table. I can be convinced that moving this to a global level is out of scope for this PR.

christian-konrad replied:
I don't know if it's helpful on other tables; just the 3 I have benefit from a condensed layout since it would require the reader to scroll too much otherwise, and hence lose overview of the whole table. But generally I think docusaurus tables are a bit too spacy-but don't want to judge by my personal taste

pepopowitz reviewed (commented):
This is still marked draft so I'm not sure how much of it is still in progress, but I have a couple blocking comments.

christinaausley commented on docs/guides/assets/api-architecture.png:
This makes me think, I'd really like to review diagrams before they're PRed. For example, "Job worker-based tasks" and "Zeebe-based tasks".

This isn't blocking, but I should create an issue in the backlog to determine how I can review these diagrams before they're saved and PRed.

christian-konrad replied:
Good catch, I sentence-cased it and fixed the hyphenation

christinaausley commented on docs/guides/migrate-to-zeebe-user-tasks.md:
```diff
@@ -0,0 +1,376 @@
+---
+id: migrate-to-zeebe-user-tasks
+title: Migrate to Zeebe user tasks
+description: "Migrating user tasks from job-based to Zeebe implementation"
```
Maybe "Learn how to migrate job worker-based user tasks to Zeebe-based tasks."

christinaausley commented on docs/guides/migrate-to-zeebe-user-tasks.md:
```diff
@@ -0,0 +1,376 @@
+import ZeebeTaskSelectionImg from './assets/zeebe-user-task-selection.png';
+
+Camunda 8.5 introduces a new [user task](/docs/components/modeler/bpmn/user-tasks/user-tasks.md) implementation type: Zeebe user tasks.
+Zeebe user tasks have a couple of benefits, including running directly on the automation engine, removing dependencies and round trips to the Tasklist web app, and a more powerful API.
```
I would probably just say "round trips to Tasklist".

christinaausley commented on docs/guides/migrate-to-zeebe-user-tasks.md:
```diff
@@ -0,0 +1,376 @@
+import ZeebeTaskSelectionImg from './assets/zeebe-user-task-selection.png';
+
+Camunda 8.5 introduces a new [user task](/docs/components/modeler/bpmn/user-tasks/user-tasks.md) implementation type: Zeebe user tasks.
+Zeebe user tasks have a couple of benefits, including running directly on the automation engine, removing dependencies and round trips to the Tasklist web app, and a more powerful API.
```
"a more powerful API" -- would explicitly say how so.

christian-konrad replied:
Agree

christinaausley commented on docs/guides/migrate-to-zeebe-user-tasks.md:
```diff
@@ -0,0 +1,376 @@
+        </td>
+        <td>
+            <YesItem />
+            <TableTextSmall>Full lifecycle events incl. custom actions</TableTextSmall>
```
by incl. and excl. do you mean including and excluding? I would probably just write it out in full.

christinaausley commented on docs/guides/migrate-to-zeebe-user-tasks.md:
```diff
@@ -0,0 +1,376 @@
+Repeat these steps for all user tasks in the process. Finally, deploy the process to your dev cluster and test it, e.g. run the process and ensure your custom task applications work.
+
+:::caution
+Make sure to migrate from embedded to linked forms first before changing the implementation type. Use the undo function in case you accidently removed your embedded form.
```
By "undo function" do you mean "icon"?

christian-konrad replied:
No, the key shortcut—let me write that out :)

christinaausley commented on docs/guides/migrate-to-zeebe-user-tasks.md:
```diff
@@ -0,0 +1,376 @@
+        <td><span style={{ color: "green" }}>✔</span> Zeebe tasks</td>
+    </tr>
+    <tr>
+        <th style={{ textAlign: "end" }}>Safe + retrieve draft variables</th>
```
Do you mean "save and retrieve draft variables"?

christinaausley commented on docs/guides/migrate-to-zeebe-user-tasks.md:
```diff
@@ -0,0 +1,376 @@
+        </td>
+    </tr>
+    <tr>
+        <th style={{ textAlign: "end" }}>Safe + retrieve draft variables</th>
```
Same comment here.

christinaausley commented:
@christian-konrad is this still planned for release tomorrow? Let me know if I can assist with getting this merged if so.

christian-konrad commented:
@christinaausley yes, sorry there was internet outage today, I will implement your comments in some minutes :)

christian-konrad commented:
@christinaausley I incorporated your feedback and moved it another space, since this is not relevant to any new user. You will now find it under APIs & Tools.

<img width="295" alt="Bildschirmfoto 2024-03-11 um 20 57 51" src="https://github.com/camunda/camunda-docs/assets/56470362/7ad1b92b-463a-4abd-ac77-d115db0935d7">
